restcomm/restcomm.sms/src/main/java/org/restcomm/connect/sms/smpp/SmppMessageHandler.java

```diff
@@ -129,6 +132,20 @@ public void onReceive(Object message) throws Exception {
                 logger.info("SmppMessageHandler processing Outbound Message " + message.toString());
             }
             outbound((SmppOutboundMessageEntity) message);
+        } else if (message instanceof SmppDeliveryReceiptEntity){
+            SmppDeliveryReceiptEntity smppDeliveryReceiptEntity = (SmppDeliveryReceiptEntity)message;
+            SmsMessagesDao smsMessagesDao = storage.getSmsMessagesDao();
+            SmsMessage smsMessage = smsMessagesDao.getSmsMessage(smppDeliveryReceiptEntity.getDaoMessageSid());
+            SmsMessage.Status status = smppDeliveryReceiptEntity.getStatus();
+            DateTime dateSent =smppDeliveryReceiptEntity.getDateSent();
+
+            if(logger.isInfoEnabled()) {
+                logger.info("updating "+smsMessage.getSid()+" status="+status+" dateSent="+dateSent);
+            }
+
+            if(smsMessage != null && status != null) {
+                smsMessagesDao.updateSmsMessage(smsMessage.setStatus(status).setDateSent(dateSent));
```

Comment:
is this dateSent, or dateDelivered? not sure if we have a particular Date field for this...